src/patternfly/components/Table/table.scss

```diff
@@ -432,26 +432,23 @@
 // Compact table
 // ==================================================================
 .pf-c-table.pf-m-compact {
+  &:not(.pf-m-border-rows) {
+    tbody {
+      --pf-c-table--BorderWidth: 0;
+      --pf-c-table--BorderColor: transparent;
+
+      tr:first-child > * {
+        --pf-c-table-cell--PaddingTop: calc(var(--pf-c-table--m-compact-cell--PaddingTop) * 2);
+      }
+    }
+  }
+
   // Thead only
   thead {
     --pf-c-table--m-compact-cell--PaddingTop: var(--pf-c-table--m-compact-th--PaddingTop);
     --pf-c-table--m-compact-cell--PaddingBottom: var(--pf-c-table--m-compact-th--PaddingBottom);
   }
 
-  // Tbody only
-  tbody {
-    --pf-c-table--BorderWidth: 0;
-    --pf-c-table--BorderColor: transparent;
-
-    tr:first-child > * {
-      --pf-c-table-cell--PaddingTop: calc(var(--pf-c-table--m-compact-cell--PaddingTop) * 2);
-    }
-
-    tr:last-child > * {
-      --pf-c-table-cell--PaddingBottom: calc(var(--pf-c-table--m-compact-cell--PaddingBottom) * 2);
```

Comment:
What was the reason for removing this? I think now when we insert a compact table the spacing will be off like in the compound expansion:

<img width="953" alt="Screen Shot 2019-03-14 at 9 13 10 AM" src="https://user-images.githubusercontent.com/20118816/54359991-33a4da00-463a-11e9-9eae-e2ece2d824a5.png">

Reply:
@mceledonia had asked to remove the extra padding on the last row, but maybe he meant only when we apply borders? I added this space back to the default compact table, and am removing it when we have borders (along with the extra top padding on the first row).

Here's what it looks like now. WDYT?

<img width="966" alt="Screen Shot 2019-03-14 at 8 58 26 AM" src="https://user-images.githubusercontent.com/35148959/54362720-70bb9d00-4637-11e9-952b-7f465b59e3fc.png">
<img width="958" alt="Screen Shot 2019-03-14 at 8 58 17 AM" src="https://user-images.githubusercontent.com/35148959/54362721-70bb9d00-4637-11e9-9bbd-539ab3b3e336.png">

Reply:
@mcoker Much better, you nailed it. Only need that bottom padding removed when we use borders because the borders create a visual height for each row, and not removing it makes the bottom row look awkwardly larger than the rest. But we do want it without the borders because sort of the opposite happens :)

Reply:
Thanks @mceledonia! @christiemolloy updated that.